packages/esm-patient-orders-app/src/order-cancellation-form/cancel-order-form.component.tsx

```diff
@@ -0,0 +1,182 @@
+      (res) => {
+        setIsSubmitting(false);
+        closeWorkspace();
+        mutate();
+
+        showSnackbar({
+          isLowContrast: true,
+          title: t('cancelOrder', 'Cancel Order'),
+          kind: 'success',
+          subtitle: t(
+            'successfullyCancelledOrder',
+            'You have successfully cancelled an order with OrderNumber {{orderNumber}}',
+            { orderNumber: order.orderNumber },
+          ),
+        });
+      },
+      (err) => {
+        setIsSubmitting(false);
+        showSnackbar({
+          isLowContrast: true,
+          title: t(`errorCancellingOrder', 'Error cancelling order`),
+          kind: 'error',
+          subtitle: err?.message,
+        });
+      },
+    );
+  }, []);
+
+  return (
+    <Form className={styles.form}>
+      <div className={styles.grid}>
+        <Stack>
+          <section>
+            <h4 style={{ marginBottom: '1rem' }}>{order.display}</h4>
+          </section>
+          <section>
+            <Controller
+              name="cancellationDate"
+              control={control}
+              render={({ field: { onChange, value } }) => (
+                <div className={styles.row}>
+                  <DatePicker
+                    id="vaccinationExpiration"
+                    className="vaccinationExpiration"
+                    minDate={dayjs().startOf('day').format('DD/MM/YYYY')}
+                    dateFormat="d/m/Y"
+                    datePickerType="single"
+                    value={value}
+                    onChange={([date]) => onChange(date)}
+                    autocomplete="off"
+                  >
+                    <DatePickerInput
+                      id="date-picker-calendar-id"
+                      placeholder="dd/mm/yyyy"
+                      labelText={t('cancellationDate', 'Cancellation date')}
+                      type="text"
+                    />
+                  </DatePicker>
+                </div>
+              )}
+            />
+          </section>
+          <section>
+            <Controller
+              name="reasonForCancellation"
+              control={control}
+              render={({ field: { onChange, value } }) => (
+                <div className={styles.row}>
+                  <TextArea
+                    type="text"
+                    id="reasonForCancellation"
+                    labelText={t('reasonForCancellation', 'Reason for cancellation')}
+                    value={value}
+                    onChange={(evt) => onChange(evt.target.value)}
+                  />
+                </div>
+              )}
+            />
+          </section>
+        </Stack>
+      </div>
+
+      {showErrorNotification && (
+        <Column className={styles.errorContainer}>
+          <InlineNotification
+            lowContrast
+            title={t('error', 'Error')}
+            subtitle={t('pleaseFillField', 'Please fill at least one field') + '.'}
```

Comment:
@pirupius i think this message is confusing and might create date inconsistencies... why at least one field ? Is the requirement that any of the two fields should be filled in order for the form to be considered valid?

Reply:
Will make changes to this

Reply:
I agree, it can say, please fill the required fields.